CMakeLists.txt

```diff
@@ -32,7 +32,16 @@ set(${PROJECT_NAME}_SRCS
   src/meta_object.cpp
   src/multi_library_class_loader.cpp
 )
+
+
+set(SHARED_BUILD false CACHE BOOL "Build shared version of library")
```

Comment:
There is a CMake variable for exactly this behavior (http://www.cmake.org/cmake/help/v2.8.12/cmake.html#variable:BUILD_SHARED_LIBS) so I am not sure why to introduce this custom variable?